drive/src/query/mod.rs

```diff
@@ -946,4 +1002,369 @@ mod tests {
+            "limit": 100,
+            "orderBy": [
+                ["firstName", "asc"],
+                ["lastName", "asc"],
+            ],
+        });
+
+        let contract = Contract::default();
+        let document_type = DocumentType::default();
+
+        let where_cbor = common::value_to_cbor(query_value, None);
+        let query = DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &document_type)
+            .expect_err("the query should not be created");
+    }
+
+    #[test]
+    fn test_valid_query_groupable_meeting_clauses() {
+        let query_value = json!({
+            "where": [
+                ["firstName", "<=", "Gilligan"],
+                ["firstName", ">", "Gilligan"],
+            ],
+            "limit": 100,
+            "orderBy": [
+                ["firstName", "asc"],
+                ["lastName", "asc"],
+            ],
+        });
+
+        let contract = Contract::default();
+        let document_type = DocumentType::default();
+
+        let where_cbor = common::value_to_cbor(query_value, None);
+        let query = DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &document_type)
+            .expect("the query should be created");
+    }
+
+    #[test]
+    fn test_valid_query_query_field_at_max_length() {
+        let long_string = "t".repeat(255);
+        let query_value = json!({
+            "where": [
+                ["firstName", "<", long_string],
+            ],
+            "limit": 100,
+            "orderBy": [
+                ["firstName", "asc"],
+                ["lastName", "asc"],
+            ],
+        });
+        let contract = Contract::default();
+        let document_type = DocumentType::default();
+
+        let where_cbor = common::value_to_cbor(query_value, None);
+        DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &document_type)
+            .expect("query should be fine for a 255 byte long string");
+    }
+
+    #[test]
+    fn test_invalid_query_field_too_long() {
+        let (drive, contract) = setup_family_contract();
+
+        let document_type = contract
+            .document_type_for_name("person")
+            .expect("expected to get a document type");
+
+        let too_long_string = "t".repeat(256);
+        let query_value = json!({
+            "where": [
+                ["firstName", "<", too_long_string],
+            ],
+            "limit": 100,
+            "orderBy": [
+                ["firstName", "asc"],
+                ["lastName", "asc"],
+            ],
+        });
+
+        let where_cbor = common::value_to_cbor(query_value, None);
+        let query = DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &document_type)
+            .expect("fields of queries length must be under 256 bytes long");
+        query
+            .execute_no_proof(&drive, None)
+            .expect_err("fields of queries length must be under 256 bytes long");
+    }
+
+    // Eventually we want to error with weird Null values
```

Comment:
Add "TODO"

Reply:
There's no point, as we write all queries at some point in Rust. There's a massive todo here.